crates/shared/src/sources/balancer_v2/pool_storage.rs

```diff
@@ -178,12 +181,16 @@ impl PoolStorage {
         let weighted_pools = construct_pool_map(initial_weighted_pools, &mut pools_by_token);
         let stable_pools = construct_pool_map(initial_stable_pools, &mut pools_by_token);
 
-        PoolStorage {
+        let mut storage = PoolStorage {
             pools_by_token,
             weighted_pools,
             stable_pools,
             data_fetcher,
-        }
+            initial_fetched_block: 0,
+        };
+        storage.initial_fetched_block = storage.last_event_block();
```

Comment:
nit: Looks a bit like this should be part of the constructor logic (rather than relying on the caller to do that)?

Reply:
This is in the constructor. We went from:
```
return PoolStorage { ... }
```

to:
```
let mut storage = PoolStorage { ... }
storage.initial_fetched_block = ...
return storage
```

The caller doesn't need to do anything. The reason I wrote it in this sort of icky way is to be able to reuse the `PoolStorage::last_event_block` function. This implementation detail shouldn't leak to the caller though.

Reply:
I'm going ahead to merge this so that we can try it out in staging before using it in prod. Let me know if I misunderstood the nit and I will fix it in a follow up.